src/CheckSessionIFrame.ts

```diff
@@ -52,9 +54,13 @@ export class CheckSessionIFrame {
         ) {
             if (e.data === "error") {
                 this._logger.error("error message from check session op iframe");
-                if (this._stopOnError) {
+                if (this._stopOnError && !this.propagateUserSessionError) {
```

Comment:
if `_stopOnError` is true i really would expect that `this.stop()` should be called. You need to rearrange your code changes